### PR TITLE
[MOV] acc_pay_patch->acc_pay: merge bugfix module

### PR DESCRIPTION
**Pull request by Feyensv**
The module ´account_payment_invoice_online_payment_patch´ was added in stable to allow `account_payment` users to disable online invoice payment (without removing the module, critical for other flows).

It can be safely merged into the `account_payment` module for future versions.

Original bugfix PR: https://github.com/odoo/odoo/pull/110177
Related to opw-3114860

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Feyensv** [comment]
waiting for PO input to simplify/clarify this whole binz

**Comment by AntoineVDV** [comment]
Link to original commit/task?

**Comment by Feyensv** [comment]
Added on PR message for now (can push it too if you prefer, wasn't sure if you'd prefer one or two commits)

File: addons/account_payment/views/account_portal_templates.xml [hunk 1]
```diff
@@ -16,23 +14,41 @@
         <xpath expr="//t[@t-foreach='invoices']/tr/td[hasclass('tx_status')]" position="replace">
             <t t-set="last_tx" t-value="invoice.get_portal_last_transaction()"/>
             <td class="tx_status text-center">
-                <t t-if="invoice.state == 'posted' and invoice.payment_state in ('not_paid', 'partial') and (last_tx.state not in ['pending', 'authorized', 'done', 'cancel'] or (last_tx.state == 'pending' and last_tx.provider_code in ('none', 'custom')))">
-                    <span class="badge rounded-pill text-bg-info"><i class="fa fa-fw fa-clock-o"></i><span class="d-none d-md-inline"> Waiting for Payment</span></span>
+                <t t-if="last_tx">
+                    <!-- c/p of account_payment -->
```
Comment (AntoineVDV) [inline addons/account_payment/views/account_portal_templates.xml]: You can drop the comment now.

File: addons/account_payment/models/account_move.py [hunk 2]
```diff
@@ -39,8 +40,12 @@ def _compute_amount_paid(self):
             )
 
     def _has_to_be_paid(self):
+        # FIXME ANVFE do we only wanna consider last tx or all txs?
+        # In portal template we consider last tx sometimes, all txs sometimes....
```
Comment (AntoineVDV) [inline addons/account_payment/models/account_move.py:44]: All. Let's be consistent with the model-object, especially if it was somewhat supported before.

File: addons/account_payment/models/account_move.py [hunk 3]
```diff
@@ -51,7 +56,8 @@ def _has_to_be_paid(self):
                 self.amount_residual
                 # FIXME someplace we check amount_residual and some other amount_paid < amount_total
                 # what is the correct heuristic to check ?
-                or not transactions
+                or not (transactions or pending_transactions)
+                # consider pending txs or not ?
```
Comment (AntoineVDV) [inline addons/account_payment/models/account_move.py:60]: I'd say yes, but we can keep the current behavior for now if it wasn't the case and rethink all portals as a whole in master.

File: addons/account_payment/views/account_portal_templates.xml (hunk 1 above)
Comment (Feyensv) [reply addons/account_payment/views/account_portal_templates.xml]: dropped by second commit (should i merge them ?)

File: addons/account_payment/models/account_move.py (hunk 2 above)
Comment (Feyensv) [reply addons/account_payment/models/account_move.py:44]: We should drop this `_get_portal_last_tx` logic sometimes then

File: addons/account_payment/models/account_move.py (hunk 3 above)
Comment (Feyensv) [reply addons/account_payment/models/account_move.py:60]: Well, that's the problem, it was somewhat considered before (see the pending_manual_txs thing in the template before).
Let's consider them for now, worst case we can change our minds if needed.

File: addons/account_payment/views/account_portal_templates.xml (hunk 1 above)
Comment (AntoineVDV) [reply addons/account_payment/views/account_portal_templates.xml]: No, it's clearer with two commits. If you feel like it you can amend the first commit :p

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**